src/dmd/dsymbolsem.d

```diff
@@ -1327,6 +1327,16 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
             else
                 dsym.edtor = dsym.edtor.expressionSemantic(sc);
 
+            // if the variable is global or static, store the destruction
+            // expression so it can be used in the static destructor of the
+            // module.
+            auto m = dsym.getModule();
+            if (dsym.isStatic || sc.parent == m)
```

Comment:
I guess you will have to distinguish between TLS and `shared/__gshared` data here.

Reply:
Ah yes, I always forget about that case.

Reply:
> You mean that shared variables should be destroyed in a shared static destructor?

Yes, that should be the case. A non-shared static destructor will run on every thread termination. If that is destroying the shared variables, then you will have problems.